src/messaging/views.py

```diff
@@ -0,0 +1,129 @@
+from django.shortcuts import render, get_object_or_404
+from django.conf import settings
+from django.db.models import Q, Max
+from rest_framework.response import Response
+from .models import Thread, Message
+from .serializers import MessageSerializer, ThreadSerializer
+from users.models import Profile, User
+from rest_framework import generics
+from rest_framework.views import APIView
+from pickmybruin.settings import REQUEST_TEMPLATE
+
+import sendgrid
+from sendgrid.helpers.mail import Email, Content, Substitution, Mail
+
+# Create your views here.
+
+class ReadMessageView(generics.UpdateAPIView):
+    """
+    View for marking a specified message as read
+    """
+    serializer_class = MessageSerializer
+
+    def patch(self, request, *args, **kwargs):
+        message_id = int(self.kwargs['message_id'])
+        message = get_object_or_404(Message, id=message_id)
+
+        message.unread = False
+
+        message.save()
+
+        return Response(MessageSerializer(message).data)
+
+
+class CheckHistoryView(APIView):
+    """
+    View for checking the existence of a thread between two users
+    """
+    def get(self, request, *args, **kwargs):
+
+        #check if thread exists
+        my_profile = get_object_or_404(Profile, user=self.request.user)
+        other_id = int(self.kwargs['profile_id'])
+        other_profile = get_object_or_404(Profile, id=other_id)
+
+        query = Q(profile_1=my_profile) & Q(profile_2=other_profile)
+        query |= Q(profile_2=my_profile) & Q(profile_1=other_profile)
```

Comment:
let's make this query stuff a helper function.  i can see this being used and reused a lot.